clients/common/clientComm.py

```diff
@@ -37,6 +37,58 @@ def stopComm(self):
         self.runComm = False
         self.wsConnected = False
 
+
+    def sendReceiveMsg(self, req, timeout): 
+        jsonDump = json.dumps(req)
+        self.sendMsgQueue.put(jsonDump)
+        while True:
+            try:
+                resp = self.recvMsgQueue.get(timeout = timeout)
+                respJson =  json.loads(resp) 
+                if str(req["requestId"]) == str(respJson["requestId"]):
+                    return resp
+            except queue.Empty:
+                req["error"] =  "timeout"
+                return json.dumps(req, indent=2) 
+
+    def authorize(self, token, timeout = None):
+        if os.path.isfile(token):
+            with open(token, "r") as f:
+                token = f.readline()
+
+        req = {}
+        req["requestId"] = 1238
+        req["action"]= "authorize"
+        req["tokens"] = token
+        return self.sendReceiveMsg(req, timeout)
+
+    def getMetaData(self, path, timeout = 1):
+        """Get MetaData of the parameter"""
+        req = {}
+        req["requestId"] = 1236
+        req["action"]= "getMetaData"
+        req["path"] = path 
+        return self.sendReceiveMsg(req, timeout)
+
+    def setValue(self, path, value, timeout = 1):
+        if 'nan' == value:
+            print(path + " has an invalid value " + str(value))
+            return
+        req = {}
+        req["requestId"] = 1235
+        req["action"]= "set"
+        req["path"] = path
+        req["value"] = value
+        return self.sendReceiveMsg(req, timeout)
+
+
+    def getValue(self, path, timeout = 5):
+        req = {}
+        req["requestId"] = 1234
```

Comment:
This means, it is still broken in testclient. To provoke it: Reduce timeout to a small value and try this:

```
pipenv run python testclient.py 
connect to wss://127.0.0.1:8090
Websocket connected!!
VSS Client> authorize ../../certificates/jwt/super-admin.json.token 
{
    "TTL": 1767225599, 
    "action": "authorize", 
    "requestId": "1238", 
    "timestamp": 1607118379
}

VSS Client> getValue Vehicle
{
  "requestId": 1234,
  "action": "get",
  "path": "Vehicle",
  "error": "timeout"
}


VSS Client> getValue Vehicle.Speed
[...]
        "Vehicle.VehicleIdentification.vehicleModelDate": "---", 
        "Vehicle.VehicleIdentification.vehicleSeatingCapacity": "---", 
        "Vehicle.VehicleIdentification.vehicleSpecialUsage": "---", 
        "Vehicle.VehicleIdentification.vehicleinteriorColor": "---", 
        "Vehicle.VehicleIdentification.vehicleinteriorType": "---", 
        "Vehicle.accelerationTime": "---", 
        "Vehicle.cargoVolume": "---", 
        "Vehicle.emissionsCO2": "---"
    }
}

VSS Client> 
```
So when the first one times out, the id comparison can not rescue you, because everything has the same id